core/server/master/src/test/java/alluxio/master/file/meta/UfsAbsentPathCacheTest.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file.meta;
+
+import alluxio.AlluxioURI;
+import alluxio.master.file.options.MountOptions;
+import alluxio.underfs.MasterUfsManager;
+import alluxio.underfs.UfsManager;
+import alluxio.util.IdUtils;
+
+import com.google.common.io.Files;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.util.Collections;
+
+/**
+ * Unit tests for {@link UfsAbsentPathCache}.
+ */
+public class UfsAbsentPathCacheTest {
```

Comment:
so we have a unit test for the sync cache, but use the async cache in production?